main.py

```diff
@@ -1511,43 +1512,90 @@ def show_hover(self, point, contents):
             max_width=800)
 
 
-class CompletionHandler(sublime_plugin.EventListener):
-    def __init__(self):
-        self.completions = []  # type: List[Tuple[str, str]]
-        self.refreshing = False
+class CompletionState(object):
```

Comment:
I had some trouble seeing "old" completions when I typed fast, I hope this class fixes that.

Reply:
From some quick tests using the changes in this PR I don't see this problem anymore, nice job!